anchore_engine/services/policy_engine/engine/tasks.py

```diff
@@ -671,8 +671,9 @@ def execute(self) -> bool:
         rtype: bool
         """
         try:
-            return GrypeDBSyncManager.run_grypedb_sync(self.grypedb_file_path)
-        except NoActiveGrypeDB:
+            session = get_session()
+            return GrypeDBSyncManager.run_grypedb_sync(session, self.grypedb_file_path)
```

Comment:
Any reason why context manager can't be used here? That'll handle transaction commit/rollback and close the session 
`with session_scope() as session:`